### PR TITLE
Fix GPG email checking to be case insensitive

### LINKED ISSUE #1986: E-Mail verifcation is case sensitive but all emails are added lowercase
- Gitea version (or commit ref):
- Git version: 2.1.4
- Operating system:
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [X] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [X] Yes (provide example URL)
  - [ ] No
  - [ ] Not relevant
- Log gist: Not relevant

## Description
I can add secondary E-Mail Addresses to my account. These are all lowercase in my profile no matter how I type them by adding them.
If I try to add a PGP Public key E-Mail verification is case sensitive. In my case it's two uppercase letters like Max@Musterman.com. Although the E-Mail is added to the profile and confirmed I still have the same error complaining 'The email attached to the GPG key couldn't be found or is not confirmed yet: Max@Musterman.com[0]'

[0] Mail Address changed

### PR DESCRIPTION
**Pull request by lafriks**
Fix #1986

### CODE REVIEW

File: models/gpg_key.go [hunk 1]
```diff
@@ -213,7 +213,7 @@ func parseGPGKey(ownerID int64, e *openpgp.Entity) (*GPGKey, error) {
 	for _, ident := range e.Identities {
 
 		for _, e := range userEmails {
-			if e.Email == ident.UserId.Email && e.IsActivated {
+			if e.Email == strings.ToLower(strings.TrimSpace(ident.UserId.Email)) && e.IsActivated {
```
Comment (sapk) [inline models/gpg_key.go]: For little performance gain maybe cache strings.ToLower(strings.TrimSpace()) result in a tmp variable and not do it at each userEmails.

**Comment by sapk** [comment]
A little performance remark but otherwise LGTM.

**Comment by lafriks** [comment]
Fixed suggested changes and added unit test

**Comment by lunny** [comment]
LGTM

**Pull request merged**